source/common/ssl/context_impl.cc

```diff
@@ -209,7 +209,7 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
         ASN1_STRING* str = altname->d.uniformResourceIdentifier;
         char* crt_san = reinterpret_cast<char*>(ASN1_STRING_data(str));
```

Comment:
Yes, that is correct. Optimally, You would just pass char* and length to the function, then still use strncmp (or compare if you can). At this point though let's just get something working that everyone agrees with. :)